src/WinRT.Runtime/Projections/Bindable.net5.cs

```diff
@@ -396,22 +398,82 @@ internal unsafe interface IEnumerable : global::System.Collections.IEnumerable,
-                    _enumerator = (IWinRTObject obj) => (global::System.Collections.IEnumerator)getEnumerator.Invoke(obj, null);
+                Type enumGenericType;
+
+                // First, look for and get the IEnumerable<> interface implemented by this type, if one exists. The scenario is, imagine you
+                // got an 'IList<string>' from somewhere, and then you use LINQ with it. What LINQ does is it converts both to object. Then
+                // it does a cast to 'IEnumerable'. At this point, you are trying an IDIC cast for 'IEnumerable' and asking the IDIC 'IEnumerable'
+                // interface to handle it. The question to answer is, what version of 'IEnumerable' is it. Is it the one provided by 'IList<string>'
+                // or is it the one that maps to 'IBindableIterable'. We actually don't know at this point which one to use, especially given the
+                // 'IBindableIterable' interface may not even be implemented on the native object, as it was an 'IList<string>' originally. This is
+                // also why we can't just check whether the object implements 'IEnumerable' and just call 'GetEnumerator()' on that.
+                if (runtimeType.IsGenericType && runtimeType.GetGenericTypeDefinition() == typeof(IEnumerable<>))
+                {
+                    enumGenericType = runtimeType;
                 }
+                else
+                {
+                    [SuppressMessage("Trimming", "IL2070", Justification =
+                        """
+                        'SomeType.GetInterfaces().Any(t => t.GetGenericTypeDefinition() == typeof(IEnumerable<>)' is safe,
+                        provided you obtained someType from something like an analyzable 'Type.GetType' or 'object.GetType'
+                        (i.e. it is safe when the type you're asking about can exist on the GC heap as allocated).
+                        """)]
+                    [MethodImpl(MethodImplOptions.NoInlining)]
+                    static Type GetEnumerableOfTInterface(Type runtimeType)
+                    {
+                        foreach (Type interfaceType in runtimeType.GetInterfaces())
+                        {
+                            if (interfaceType.IsGenericType && interfaceType.GetGenericTypeDefinition() == typeof(IEnumerable<>))
+                            {
+                                return interfaceType;
+                            }
+                        }
+
+                        return null;
+                    }
+
+                    enumGenericType = GetEnumerableOfTInterface(runtimeType);
+                }
+
+#if NET8_0_OR_GREATER
+                var methodInfo = (MethodBase)enumGenericType?.GetMemberWithSameMetadataDefinitionAs(EnumerableOfTGetEnumerator);
+
+                _enumerator = methodInfo is null ? null : MethodInvoker.Create(methodInfo);
+#else
+                _enumerator = enumGenericType?.GetMethod("GetEnumerator");
```

Comment:
Any reason you didn't use `GetMemberWithSameMetadataDefinitionAs` on the else case here on NET 6?  I assume we can cast to MethodBase and be able to call invoke on it.